pkgs/build-support/trivial-builders.nix

```diff
@@ -300,6 +300,9 @@ rec {
    *     |       `-- stack.fish -> /nix/store/6lzdpxshx78281vy056lbk553ijsdr44-stack-2.1.3.1/share/fish/vendor_completions.d/stack.fish
    * ...
    *
+   * # creates multiple outputs, each the result of combining the corresponding outputs of the paths
```

Comment:
This comment is a bit imprecise as, like before, the `out` output will contain the primary output (first item in `outputs` list) when passed package attribute. Or even specific output when passed that.

Try:

```ShellSession
$ nix-build -E 'with import ./. {}; symlinkJoin { name = "foo"; paths = [ glib.dev ]; }'
$ nix-build -E 'with import ./. {}; symlinkJoin { name = "foo"; paths = [ glib ]; }'
```

Reply:
The comment refers to the example given just below it, not generally.

Reply:
Maybe that could be made more obvious, though. I was trying to follow the structure of the examples above it..

Reply:
Even then, you will end up with `openssl.bin` in `out`, not `openssl.out`.

Reply:
Ah, yes, now I see what you mean. I've updated the comment to reflect this.

I would guess that we want the main outputs to be combined like this almost always, but if there are times where it's inappropriate, we a parameter to change the behavior. Do you see a use-case where we need the outputs to correspond exactly?